File: pages/BasePage.ts
import { type Page } from '@playwright/test';

/**
 * Base class for all Page Objects.
 *
 * Provides the single shared navigateTo() implementation so every page
 * does the same thing on load: go to URL, wait for networkidle.
 * Subclasses override the public navigateTo() with their specific path.
 */
export class BasePage {
  constructor(protected readonly page: Page) {}

  protected async navigateTo(path: string): Promise<void> {
    await this.page.goto(path);
    await this.page.waitForLoadState('networkidle');
  }
}

File: pages/base_page.py
from playwright.sync_api import Page


class BasePage:
    """
    Base class for all Page Objects.
    Provides a shared navigate_to() so every page does the same thing on load:
    go to path, wait for networkidle.
    """

    def __init__(self, page: Page):
        self.page = page

    def navigate_to(self, path: str) -> None:
        self.page.goto(path)
        self.page.wait_for_load_state("networkidle")
